### PR TITLE
feat: implements AppMemoryUsage and AppMemoryUsageLimit for Android

### LINKED ISSUE #5891: Please add MemoryManager
## What would you like to be added:
MemoryManager, at least .AppMemoryUsage and .AppMemoryUsageLimit

## Why is this needed:
I'm using it in my app (and I'm porting app to Uno).

## For which Platform:

- [ ] iOS
- [X] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by Hudney**
GitHub Issue: closes #5891

## PR Type
Feature 

## What is the current behavior?
Not implemented, throws an exception. #5891

## What is the new behavior?
Returns the correct values.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UWP/Generated/3.0.0.0/Windows.System/MemoryManager.cs [hunk 1]
```diff
@@ -27,15 +40,26 @@ public static ulong AppMemoryUsage
...
+		public static ulong AppMemoryUsageLimit
+		{
+			get
+			{
+				ActivityManager.MemoryInfo memoryInfo = new ActivityManager.MemoryInfo();
+				ActivityManager.FromContext(ContextHelper.Current)?.GetMemoryInfo(memoryInfo);
+
+				return AppMemoryUsage + (ulong) memoryInfo.AvailMem - (ulong) memoryInfo.Threshold;
+			}
+		}
```
Comment (jeromelaban) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.System/MemoryManager.cs:62]: We generally remove the code from the "generated" folder, as this code is generated by a tool. You'll add the code in another file named `MemoryManager.android.cs`, out of the generated folder, similar to this one. Only add this property's implementation to the file.

**Review by jeromelaban — COMMENTED** [review]
Thanks for your contribution!

Can you also add a new page in the SampleApp that includes a demonstration of this ?

Also, I'm wondering if we should be using the .NET `GC` APIs instead, as they give similar information, though maybe just the managed part of it.

File: src/Uno.UWP/Generated/3.0.0.0/Windows.System/MemoryManager.cs (hunk 1 above)
Comment (Youssef1313) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.System/MemoryManager.cs:62]: I don't think this is the correct place for this. Should be in a separate `.Android.cs` file

File: src/Uno.UWP/Generated/3.0.0.0/Windows.System/MemoryManager.cs [hunk 2]
```diff
@@ -1,21 +1,34 @@
...
 			{
 				throw new global::System.NotImplementedException("The member ulong MemoryManager.AppMemoryUsage is not implemented in Uno.");
 			}
 		}
+		#else
+		public static ulong AppMemoryUsage
+		{
+			get
+			{
+				return (ulong) GC.GetTotalMemory(false);
```
Comment (pkar70) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.System/MemoryManager.cs:29]: Are you sure that GC.GetTotalMemory(false) is equivalent of MemoryManager.AppMemoryUsage?
It seems to me that GC.GetTotalMemory returns number of bytes currently allocated in managed memory, i.e. bytes for data; and MemoryManager.AppMemoryUsage is data and code.

File: src/SamplesApp/UITests.Shared/Windows_System/MemoryManagerTests.xaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	mc:Ignorable="d xamarin">
+
+	<xamarin:Grid>
+		<Grid.RowDefinitions>
+			<RowDefinition Height="Auto" />
+			<RowDefinition Height="*" />
+		</Grid.RowDefinitions>
+
+		<StackPanel Background="Honeydew">
+			<TextBlock x:Name="ValueMemoryManager" Text="Click to display AppMemoryUsage and AppMemoryUsageLimit values"/>
```
Comment (jeromelaban) [inline src/SamplesApp/UITests.Shared/Windows_System/MemoryManagerTests.xaml:18]: Let's use individual fields, it's easier to deal with when working with [UI Tests](https://platform.uno/docs/articles/contributing/guidelines/creating-tests.html).

```suggestion
			<TextBlock x:Name="AppMemoryUsageResult" Text="N/A"/>
			<TextBlock x:Name="AppMemoryUsageLimitResult" Text="N/A"/>
```

You don't need a UI test for now, but can you add a runtime test that validates that the values are not zero, for instance ? You can add it here: https://github.com/unoplatform/uno/tree/master/src/Uno.UI.RuntimeTests/Tests/Windows_System

File: src/Uno.UWP/System/MemoryManager.Android.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+{
+	partial class MemoryManager
+	{
+		public static ulong AppMemoryUsage
+		{
+			get
+			{
+				return (ulong) GC.GetTotalMemory(false);
+			}
+		}
```
[the remark this answers is not in the document]
Comment (teras) [reply src/Uno.UWP/System/MemoryManager.Android.cs:15]: If we are talking about Android, the "correct" way to do it (since it's a Unix OS) is by using `getrusage` instead. This method also considers shared memory, where the documentation of `MemoryManager.AppMemoryUsage` is not clear if it includes it or not.

[the remark this answers is not in the document]
Comment (jeromelaban) [reply src/Uno.UWP/System/MemoryManager.Android.cs:15]: I think at this point shared memory is not very large great concern; we can stay on the java-android apis for now.
